crypto/storage/vault.go

```diff
@@ -0,0 +1,118 @@
+package storage
+
+import (
+	"crypto"
+	"fmt"
+	vault "github.com/hashicorp/vault/api"
+	"github.com/nuts-foundation/nuts-node/crypto/util"
+)
+
+const privateKeyPath = "nuts-private-keys"
+const kvEnginePath = "kv"
+const keyName = "key"
+
+type logicaler interface {
+	Read(path string) (*vault.Secret, error)
+	Write(path string, data map[string]interface{}) (*vault.Secret, error)
+}
+
+type vaultKVStorage struct {
+	client logicaler
+}
+
+// NewVaultKVStorage creates a new Vault backend using the kv version 1 secret engine: https://www.vaultproject.io/docs/secrets/kv
+// It currently only supports token authentication which should be provided by the token param.
+// If vaultAddr is empty, the VAULT_ADDR environment should be set.
```

Comment:
This dual possibility of configuring is confusing. VAULT_ADDR and NUTS_CRYPTO_ADDR are the same? Would be clearer if only NUTS_ type config is used?

Or is this the result of default vault params injection in cloud environments? if so this should deserve a mention in the docs.

Reply:
Yes, this desirable behavior since Nomad and other orchestrator set this env var. It is mentioned in the production configuration docs.